x-pack/plugins/apm/public/components/routing/templates/mobile_service_template/index.tsx

```diff
@@ -0,0 +1,209 @@
+import { getAlertingCapabilities } from '../../../alerting/utils/get_alerting_capabilities';
+import { SearchBar } from '../../../shared/search_bar';
+import { ServiceIcons } from '../../../shared/service_icons';
+import { TechnicalPreviewBadge } from '../../../shared/technical_preview_badge';
+import { ApmMainTemplate } from '../apm_main_template';
+import { AnalyzeDataButton } from '../apm_service_template/analyze_data_button';
+
+type Tab = NonNullable<EuiPageHeaderProps['tabs']>[0] & {
+  key: 'overview' | 'transactions' | 'service-map' | 'alerts';
+  hidden?: boolean;
+};
+
+interface Props {
+  title: string;
+  children: React.ReactChild;
+  selectedTab: Tab['key'];
+  searchBarOptions?: React.ComponentProps<typeof SearchBar>;
+}
+
+export function MobileServiceTemplate(props: Props) {
+  return (
+    <ApmServiceContextProvider>
+      <TemplateWithContext {...props} />
+    </ApmServiceContextProvider>
+  );
+}
+
+function TemplateWithContext({
+  title,
+  children,
+  selectedTab,
+  searchBarOptions,
+}: Props) {
+  const {
+    path: { serviceName },
+    query,
+    query: { rangeFrom, rangeTo },
+  } = useApmParams('/mobile-services/{serviceName}/*');
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const router = useApmRouter();
+
+  const tabs = useTabs({ selectedTab });
+
+  useBreadcrumb(
+    () => ({
+      title,
+      href: router.link(
+        `/mobile-services/{serviceName}/${selectedTab}` as const,
+        {
+          path: { serviceName },
+          query,
+        }
+      ),
+    }),
+    [query, router, selectedTab, serviceName, title]
+  );
+
+  return (
+    <ApmMainTemplate
+      pageHeader={{
+        tabs,
+        pageTitle: (
+          <EuiFlexGroup justifyContent="spaceBetween">
+            <EuiFlexItem>
+              <EuiFlexGroup alignItems="center">
+                <EuiFlexItem grow={false}>
+                  <EuiTitle size="l">
+                    <h1 data-test-subj="apmMainTemplateHeaderServiceName">
+                      {serviceName}
+                    </h1>
+                  </EuiTitle>
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <ServiceIcons
+                    serviceName={serviceName}
+                    start={start}
+                    end={end}
+                  />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <TechnicalPreviewBadge />
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiFlexItem>
+
+            <EuiFlexItem grow={false}>
+              <AnalyzeDataButton />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        ),
+      }}
+    >
+      <SearchBar {...searchBarOptions} />
+      <ServiceAnomalyTimeseriesContextProvider>
+        {children}
+      </ServiceAnomalyTimeseriesContextProvider>
+    </ApmMainTemplate>
+  );
+}
+
+function useTabs({ selectedTab }: { selectedTab: Tab['key'] }) {
```

Comment:
I feel like this should be called `getTabs` instead of `useTabs`

Reply:
as a custom hook shouldn't Its name always start with `use` ?